dfttopif/parsers/base.py

```diff
@@ -40,22 +40,29 @@ class DFTParser(object):
      have the same name in the pif.
     '''
     
-    _directory = None
-    '''Path to directory containing calculation files'''
-    
     _converged = None
     ''' Whether this calculation has converged '''
     
-    def __init__(self, directory):
-        '''Initialize a parser.
+    def __init__(self, files):
+        '''Initialize a parser by defining the list of files that the parser can read from.
         
         Input:
-            directory - String, path to a directory of output files
+            files - [str], list of files usable by this parser.
         Raises:
-            Par - If parser cannot find needed files
+            InvalidIngesterException - If parser cannot find needed files
         '''
+        self._files = files
+
+    @classmethod
+    def generate_parser(cls, directory):
```

Comment:
If this only works with a directory, self-document it in the method name, e.g. `generate_from_directory`.

Reply:
👍